agent/nginx_reload_manager.py

```diff
@@ -0,0 +1,213 @@
+                import traceback
+
+                traceback.print_exc()
+                self.log(f"Error during processing requests : {e!s}")
+
+    def _reload_nginx(self) -> RELOAD_STATUS_TYPE:
+        from agent.proxy import Proxy
+
+        if self._should_skip_nginx_reload():
+            return "Skipped"
+
+        try:
+            proxy = Proxy()
+            proxy._generate_proxy_config()
+            subprocess.run("sudo nginx -s reload", shell=True, check=True)
+            self.last_reload_at = datetime.now()
+            return "Success"
+        except subprocess.CalledProcessError as e:
+            match = re.search(r'conflicting parameter "(.*?)"', e.stderr)
+            domain = None
+            if match:
+                domain = match.group(1)
+            if not domain:
+                raise e
+
+            self.log(f"Domain {domain} is conflicting, attempting auto fix", print_always=True)
+
+            # Find the paths
+            paths = glob.glob(os.path.join(self.directory, f"nginx/upstreams/*/{domain}"))
+            paths.sort(key=lambda x: os.path.getmtime(x))
+
+            if len(paths) >= 2:
+                # Keep newest file and delete the rest
+                to_keep = paths[-1]
+                self.log(f"Keeping {to_keep}")
+                to_delete = paths[:-1]
+                for path in to_delete:
+                    with contextlib.suppress(FileNotFoundError):
+                        # In case RQ worker removed the file
+                        # before we could delete it
+                        os.remove(path)
+                    self.log(f"Deleted {path}", print_always=True)
+                return "Skipped"  # Send skipped status to retry the job
+
+            raise Exception(f"Not able to auto fix the issue for domain : {domain!s}") from e
+        except Exception as e:
+            import traceback
+
+            traceback.print_exc()
+            self.log(f"Error while reloading nginx : {e!s}")
+            return "Failure"
+
+    def _should_skip_nginx_reload(self):
+        """
+        ├─983331 nginx: worker process is shutting down
+        ├─983332 nginx: worker process is shutting down
+        ├─983333 nginx: worker process is shutting down
+        ├─983334 nginx: worker process is shutting down
+        ├─983361 nginx: worker process
+        ├─983362 nginx: worker process
+        ├─983363 nginx: worker process
+        ├─983364 nginx: worker process
+        └─983369 nginx: cache manager process
+        """
+        try:
+            status = subprocess.run(
+                "sudo systemctl status nginx", capture_output=True, shell=True
+            ).stdout.decode("utf-8")
+            total_workers = status.count("nginx: worker process")
+            dying_workers = status.count("nginx: worker process is shutting down")
+            active_workers = max(1, total_workers - dying_workers)
+            return (dying_workers / active_workers) >= 2
```

Comment:
This is a ratio? 

2x ratio for old vs new was already dangerous on some servers, cap at 1x or to be safe just limit to `dying_workers > 2`.

Reply:
For now lowering the ratio to 1.
If that cause any issue, then we can do `dying_workers > 2`
Else signup can slow down.